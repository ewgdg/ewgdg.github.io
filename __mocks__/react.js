const React = jest.requireActual("react")
module.exports = {
  ...React,
  useContext: jest.fn().mockImplementation(React.useContext),
}
