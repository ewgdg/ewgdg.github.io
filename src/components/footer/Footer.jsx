/* eslint-disable react/jsx-one-expression-per-line */
import React from "react"
import EmailRoundedIcon from "@material-ui/icons/EmailRounded"
import IconButton from "@material-ui/core/IconButton"

function Footer() {
  return (
    <footer
      style={{
        textAlign: "center",
        backgroundColor: "#1a1a1a",
        padding: "10vh 6vw",
        margin: "0",
      }}
    >
      <p style={{ color: "white", fontSize: "1.5rem" }}>
        Let me show you how I solve problems.
      </p>
      <p style={{ opacity: 0.6, color: "white" }}>
        <small>
          Â© {new Date().getFullYear()} Xian Zhang, Built with{" "}
          <a href="https://www.gatsbyjs.org" style={{ color: "white" }}>
            Gatsby
          </a>
        </small>
      </p>
      <div>
        <IconButton
          color="primary"
          onClick={() => {
            window.location.href = "mailto:xian.z512^gmail.com".replace(
              "^",
              "@"
            )
          }}
        >
          <EmailRoundedIcon htmlColor="white" />
        </IconButton>
      </div>
    </footer>
  )
}

export default Footer
