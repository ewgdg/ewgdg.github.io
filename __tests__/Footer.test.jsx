import React from "react"
import { render } from "@testing-library/react"
import Footer from "../src/components/footer/Footer"

test("Footer renders correctly", () => {
  const { container } = render(<Footer />)
  expect(container.firstChild).toMatchSnapshot()
})
