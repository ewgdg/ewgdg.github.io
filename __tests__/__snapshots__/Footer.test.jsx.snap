// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer
  style="text-align: center; background-color: rgb(26, 26, 26); padding: 10vh 6vw; margin: 0px; display: block; width: 100%; max-height: 350px;"
>
  <p
    style="color: white; font-size: 1.5rem;"
  >
    Let me show you how I solve problems.
  </p>
  <p
    style="opacity: 0.6; color: white;"
  >
    <small>
      Â© 
      2020
       Xian Zhang, Built with
       
      <a
        href="https://www.gatsbyjs.org"
        rel="noopener noreferrer"
        style="color: white;"
        target="_blank"
      >
        Gatsby
      </a>
    </small>
  </p>
  <div>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          color="white"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm-.4 4.25l-7.07 4.42c-.32.2-.74.2-1.06 0L4.4 8.25c-.25-.16-.4-.43-.4-.72 0-.67.73-1.07 1.3-.72L12 11l6.7-4.19c.57-.35 1.3.05 1.3.72 0 .29-.15.56-.4.72z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          color="white"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M12.007 0C6.12 0 1.1 4.27.157 10.08c-.944 5.813 2.468 11.45 8.054 13.312.19.064.397.033.555-.084.16-.117.25-.304.244-.5v-2.042c-3.33.735-4.037-1.56-4.037-1.56-.22-.726-.694-1.35-1.334-1.756-1.096-.75.074-.735.074-.735.773.103 1.454.557 1.846 1.23.694 1.21 2.23 1.638 3.45.96.056-.61.327-1.178.766-1.605-2.67-.3-5.462-1.335-5.462-6.002-.02-1.193.42-2.35 1.23-3.226-.327-1.015-.27-2.116.166-3.09 0 0 1.006-.33 3.3 1.23 1.966-.538 4.04-.538 6.003 0 2.295-1.5 3.3-1.23 3.3-1.23.445 1.006.49 2.144.12 3.18.81.877 1.25 2.033 1.23 3.226 0 4.607-2.805 5.627-5.476 5.927.578.583.88 1.386.825 2.206v3.29c-.005.2.092.393.26.507.164.115.377.14.565.063 5.568-1.88 8.956-7.514 8.007-13.313C22.892 4.267 17.884.007 12.008 0z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          color="white"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</footer>
`;
